src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -196,140 +185,137 @@ async def _failed_handler(self, events: Sequence[EnsembleFailed]) -> None:
+            self._router_socket: zmq.asyncio.Socket = zmq_context.socket(zmq.ROUTER)
+            self._router_socket.setsockopt(zmq.LINGER, 0)
+            if self._config.server_public_key and self._config.server_secret_key:
+                self._router_socket.curve_secretkey = self._config.server_secret_key
+                self._router_socket.curve_publickey = self._config.server_public_key
+                self._router_socket.curve_server = True
+
+            # Attempt to bind the ROUTER socket
+            # self._router_socket.bind(f"tcp://*:{self._config.router_port}")
+            if self._config.router_port:
+                self._router_socket.bind(f"tcp://*:{self._config.router_port}")
+            else:
+                self._router_socket.bind(self._config.url)
             self._server_started.set()
+            print(f"ROUTER listens on {self._config.url}")
+        except zmq.error.ZMQError as e:
+            logger.error(f"ZMQ error encountered {e} during evaluator initialization")
+            print(f"ZMQ error encountered {e} during evaluator initialization")
+            raise
+        try:
             await self._server_done.wait()
-            server.close(close_connections=False)
-            if self._dispatchers_connected is not None:
-                logger.debug(
-                    f"Got done signal. {self._dispatchers_connected.qsize()} "
-                    "dispatchers to disconnect..."
+            try:
+                await asyncio.wait_for(self._dispatchers_empty.wait(), timeout=5)
+            except TimeoutError:
+                logger.warning(
+                    "Not all dispatchers were disconnected when closing zmq server!"
                 )
-                try:  # Wait for dispatchers to disconnect
-                    await asyncio.wait_for(
-                        self._dispatchers_connected.join(), timeout=20
-                    )
-                except TimeoutError:
-                    logger.debug("Timed out waiting for dispatchers to disconnect")
-            else:
-                logger.debug("Got done signal. No dispatchers connected")
-
-            logger.debug("Sending termination-message to clients...")
-
             await self._events.join()
             await self._complete_batch.wait()
             await self._batch_processing_queue.join()
             event = EETerminated(ensemble=self._ensemble.id_)
             await self._events_to_send.put(event)
             await self._events_to_send.join()
-        logger.debug("Async server exiting.")
+            try:
+                await asyncio.wait_for(self._clients_empty.wait(), timeout=5)
+            except TimeoutError:
+                logger.warning(
+                    "Not all clients were disconnected when closing zmq server!"
+                )
+            logger.debug("Async server exiting.")
+        finally:
+            try:
+                self._router_socket.close()
+                zmq_context.destroy()
```

Comment:
```suggestion
                zmq_context.destroy()
```
The destroy method should close all associated sockets automatically